routes/user.go

```diff
@@ -2413,3 +2413,81 @@ func (fes *APIServer) IsFollowingPublicKey(ww http.ResponseWriter, req *http.Req
 		return
 	}
 }
+
+type IsHodlingPublicKeyRequest struct {
+	PublicKeyBase58Check            string
+	IsHodlingPublicKeyBase58Check	string
+}
+
+type IsHodlingPublicKeyResponse struct {
+	IsHodling 		bool
+	BalanceEntry 	*BalanceEntryResponse	
+}
+
+func (fes *APIServer) IsHodlingPublicKey(ww http.ResponseWriter, req *http.Request) {
+
+	decoder := json.NewDecoder(io.LimitReader(req.Body, MaxRequestBodySizeBytes))
+	requestData := IsHodlingPublicKeyRequest{}
+	if err := decoder.Decode(&requestData); err != nil {
+		_AddBadRequestError(ww, fmt.Sprintf(
+			"IsHodlingPublicKey: Problem parsing request body: %v", err))
+		return
+	}
+
+	var userPublicKeyBytes []byte
+	var err error
+	userPublicKeyBytes, _, err = lib.Base58CheckDecode(requestData.PublicKeyBase58Check)
+	if err != nil || len(userPublicKeyBytes) != btcec.PubKeyBytesLenCompressed {
+		_AddBadRequestError(ww, fmt.Sprintf(
+			"IsHodlingPublicKey: Problem decoding user public key %s: %v",
+			requestData.PublicKeyBase58Check, err))
+		return
+	}
+
+	// Get the public key for the user to check
+	var isHodlingPublicKeyBytes []byte
+	isHodlingPublicKeyBytes, _, err = lib.Base58CheckDecode(requestData.IsHodlingPublicKeyBase58Check)
+	if err != nil || len(isHodlingPublicKeyBytes) != btcec.PubKeyBytesLenCompressed {
+		_AddBadRequestError(ww, fmt.Sprintf(
+			"IsHodlingPublicKey: Problem decoding public key to check %s: %v",
+			requestData.IsHodlingPublicKeyBase58Check, err))
+		return
+	}
+
+	var utxoView *lib.UtxoView
+	utxoView, err = fes.backendServer.GetMempool().GetAugmentedUniversalView()
+	if err != nil {
+		_AddBadRequestError(ww, fmt.Sprintf("IsHodlingPublicKey: Error getting utxoView: %v", err))
+		return
+	}
+
+	var IsHodling = false
+	var BalanceEntry *BalanceEntryResponse
+
+	pkid := utxoView.GetPKIDForPublicKey(userPublicKeyBytes)
+	youHodlMap, err := fes.GetYouHodlMap(pkid, false, utxoView)
+	if err != nil {
+		_AddBadRequestError(ww, fmt.Sprintf(
+			"IsHodlingPublicKey: Could not find HODLers for pub key: %v", requestData.PublicKeyBase58Check))
+		return
+	}
+
+	for _, hodl := range youHodlMap {
+		if hodl.CreatorPublicKeyBase58Check == requestData.IsHodlingPublicKeyBase58Check {
+			BalanceEntry = hodl
+			IsHodling = true
+			break
+		}
+	}
+
+	res := IsHodlingPublicKeyResponse{
+		IsHodling: 		IsHodling,
```

Comment:
fix spacing please.

Reply:
Fixed this and a couple of other issues.